libs/hdf-converters/src/converters-from-hdf/splunk/reverse-splunk-mapper.ts

```diff
@@ -432,48 +431,64 @@ export class FromHDFToSplunkMapper extends FromAnyBaseConverter {
     }
   }
 
-  async toSplunk(
-    config: SplunkConfig,
-    filename: string,
-    webCompatibility = false
-  ): Promise<string> {
-    let service: splunkjs.Service;
-    if (webCompatibility) {
-      service = new splunkjs.Service(new ProxyHTTP.JQueryHttp(''), config);
-    } else {
-      service = new splunkjs.Service(config);
-    }
+  async toSplunk(config: SplunkConfig, filename: string): Promise<string> {
+    const hostname = config.port
+      ? `${config.scheme}://${config.host}:${config.port}`
+      : `${config.scheme}://${config.host}:8089`;
     logger.info(
-      `Logging into Splunk Service: ${config.host} with user ${config.username}`
+      `Logging into Splunk instance at ${hostname} with user ${config.username}`
     );
-    logger.verbose('Got Execution: ' + filename);
+
+    logger.verbose(`Found designated file to transfer: ${filename}`);
     const guid = createGUID(30);
-    logger.verbose('Using GUID: ' + guid);
-    return new Promise((resolve) => {
-      if (config.username && config.password) {
-        service.login((error: any) => {
-          this.handleSplunkError(error);
-        });
+    logger.verbose(`Using GUID: ${guid}`);
+
+    // Attempt to authenticate using given credentials
+    const username = (config.username ??= '');
```

Comment:
i feel like i've seen bits and pieces of these lines spread throughout the file.  is it possible to set up some utility/factory functions to do this stuff for you that would then for ex. return an object that can be parsed out with all of the stuff you could need?

Reply:
See https://github.com/mitre/heimdall2/pull/4549#discussion_r1228426104